src/components/node-list/index.js

```diff
@@ -96,73 +103,42 @@ const NodeListProvider = ({
       onToggleTagActive(item.id, false);
     } else if (isModularPipelineType(item.type)) {
       onToggleModularPipelineActive(item.id, false);
+    } else if (isElementType(item.type) && item.id === 'parameters') {
+      // Hide parameters highlight when mouse leave parameters filter item
+      onToggleParametersActive(false);
     } else if (item.visible) {
       onToggleNodeActive(null);
     }
   };
 
-  const onSectionMouseEnter = (item) => {
-    if (isParameterType(item)) {
-      onToggleParametersActive(true);
-    }
-  };
-
-  const onSectionMouseLeave = (item) => {
-    if (isParameterType(item)) {
-      onToggleParametersActive(false);
-    }
-  };
-  const onToggleGroupChecked = (type, checked) => {
-    if (isTagType(type) || isModularPipelineType(type)) {
-      // Filter all category items if at least one item set, otherwise enable all items
-      const categoryItems = items[type] || [];
-      const someCategoryItemSet = categoryItems.some(
-        (categoryItem) => !categoryItem.unset
+  const onGroupToggleChanged = (groupType) => {
+    // Enable all items in group if none enabled, otherwise disable all of them
+    const groupItems = items[groupType] || [];
+    const groupItemsDisabled = groupItems.every(
+      (groupItem) => !groupItem.checked
+    );
+
+    if (isTagType(groupType)) {
+      onToggleTagFilter(
+        groupItems.map((item) => item.id),
+        groupItemsDisabled
+      );
+    } else if (isElementType(groupType)) {
+      onToggleTypeDisabled(
+        groupItems.reduce(
+          (o, item) => ({ ...o, [item.id]: !groupItemsDisabled }),
```

Comment:
Please use a more intuitive variable name ( such as `obj`) rather than `o` here to match the naming pattern of the codebase